src/NumberGuessingGame.jsx

```diff
@@ -1,77 +1,52 @@
-  return Math.floor(Math.random() * 100) + 1;
+    return Math.floor(Math.random() * 100) + 1;
 }
 
 const MAX_ATTEMPTS = 5;
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
+const NumberGuessingGame = () => {
+    const [numberToGuess, setNumberToGuess] = useState(getRandomNumber());
+    const [numberOfGuesses, setNumberOfGuesses] = useState(0);
+    const [latestGuess, setLatestGuess] = useState(null);
 
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
+    const handleGuess = (guess) => {
+        setLatestGuess(guess);
+        setNumberOfGuesses(numberOfGuesses + 1);
     };
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
-  }
-
-  handleGuess(guess) {
-    this.setState({
-      latestGuess: guess,
-      numberOfGuesses: this.state.numberOfGuesses + 1,
-    });
-  }
-
-  handleReset() {
-    this.setState({
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    });
-  }
-
-  render() {
-    const isCorrectGuess = this.state.latestGuess === this.state.numberToGuess;
-
-    const isGameOver =
-      isCorrectGuess || this.state.numberOfGuesses === MAX_ATTEMPTS;
+    const handleReset = () => {
+        setNumberToGuess(getRandomNumber());
+        setNumberOfGuesses(0);
+        setLatestGuess(null);
+    };
+    const isCorrectGuess = latestGuess === numberToGuess;
+    const isGameOver = isCorrectGuess || numberOfGuesses === MAX_ATTEMPTS;
 
     return (
-      <div>
-        <h2>I'm thinking of a number from 1 to 100.</h2>
-        <h2>
-          Can you guess the number I am thinking of in {MAX_ATTEMPTS} tries?
-        </h2>
-        <GuessControl onGuess={this.handleGuess} />
-        {isGameOver && (
-          <GameOver hasWon={isCorrectGuess} onReset={this.handleReset} />
-        )}
-        {!isGameOver && (
-          <GuessMessage
-            guess={this.state.latestGuess}
-            numberToGuess={this.state.numberToGuess}
-            numberOfGuesses={this.state.numberOfGuesses}
-          />
-        )}
-      </div>
+        <div>
+            <h2>I'm thinking of a number from 1 to 100.</h2>
+            <h2>
+                Can you guess the number I am thinking of in {MAX_ATTEMPTS}{' '}
+                tries?
+            </h2>
+            <GuessControl onGuess={handleGuess} />
+            {isGameOver && (
+                <GameOver hasWon={isCorrectGuess} onReset={handleReset} />
+            )}
+            {!isGameOver && (
+                <GuessMessage
+                    guess={this.state.latestGuess}
+                    numberToGuess={this.state.numberToGuess}
+                    numberOfGuesses={this.state.numberOfGuesses}
```

Comment:
the state variable doesn't exist at this moment, since you're now using functional components.